src/main/java/com/google/jenkins/plugins/storage/ClassicUpload.java

```diff
@@ -55,8 +55,8 @@ public ClassicUpload(
       // Legacy arguments for backwards compatibility
       @Deprecated @Nullable String bucketNameWithVars,
       @Deprecated @Nullable String sourceGlobWithVars) {
-    super(MoreObjects.firstNonNull(bucket, bucketNameWithVars), module);
-    this.sourceGlobWithVars = MoreObjects.firstNonNull(pattern, sourceGlobWithVars);
+    super(MoreObjects.firstNonNull(bucketNameWithVars, bucket), module);
```

Comment:
As `bucketNameWithVars` and `bucket` can not be non null at the same time this will not change any logic

Reply:
No it won´t. In the end, we try to see if the parameter name used before the naming change was still in use (configuration stored on FS < `1.5.0`) and if it's the case, then the configuration file won't have the `bucket` but `bucketNameWithVars`. After the plugin upgrade and the configuration saved, the `bucketNameWithVars` won't be in the configuration stored on FS anymore, but only `bucket`.

Reply:
so then this is definatly incorrect - we want to prefer `bucket` over `bucketNameWithVars`

Reply:
So you mean that none of the constructors arguments should be `Nullable` then?

Reply:
also https://github.com/jenkinsci/google-storage-plugin/commit/23a6d931e1c3e9d73e054dfb19cf5e5e10a9c376#  the constructor sets bucketWithVars but not `bucket`  so I really do not think this should be changed.